rafthttp/pipeline.go

```diff
@@ -128,31 +132,43 @@ func (p *pipeline) post(data []byte) (err error) {
 		}
 	}()
 
-	resp, err := p.tr.pipelineRt.RoundTrip(req)
-	done <- struct{}{}
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	b, err := ioutil.ReadAll(resp.Body)
-	if err != nil {
-		p.picker.unreachable(u)
-		return err
-	}
-	resp.Body.Close()
-
-	err = checkPostResponse(resp, b, req, p.peerID)
-	if err != nil {
-		p.picker.unreachable(u)
-		// errMemberRemoved is a critical error since a removed member should
-		// always be stopped. So we use reportCriticalError to report it to errorc.
-		if err == errMemberRemoved {
-			reportCriticalError(err, p.errorc)
+	go func() {
```

Comment:
@xiang90 i have coded a select statement after this so that after cancel the program exits. Without this the test case was failing on using request.WithCancel.